src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -20,7 +20,7 @@ public UserDaoImpl(SessionFactory factory) {
     public Optional<User> findByEmail(String email) {
         try (Session session = factory.openSession()) {
             Query<User> findByEmail = session.createQuery(
-                    "FROM User WHERE email = :email", User.class);
+                    "FROM User u join fetch u.roles WHERE u.email = :email", User.class);
```

Comment:
Checklist item: 
Don’t use left join with user and roles on the dao layer. It is impossible to get user without role. Better use inner join.